src/main/scala/fi/oph/koski/opiskeluoikeus/OpiskeluoikeusChangeValidator.scala

```diff
@@ -1,16 +1,50 @@
 package fi.oph.koski.opiskeluoikeus
 
+import fi.oph.koski.eperusteet.EPerusteetRepository
 import fi.oph.koski.http.{HttpStatus, KoskiErrorCategory}
+import fi.oph.koski.organisaatio.OrganisaatioRepository
 import fi.oph.koski.schema.KoskeenTallennettavaOpiskeluoikeus
+import fi.oph.koski.validation.AmmatillinenValidation.validateVanhanOpiskeluoikeudenTapaukset
 
-object OpiskeluoikeusChangeValidator {
+class OpiskeluoikeusChangeValidator(organisaatioRepository: OrganisaatioRepository, ePerusteet: EPerusteetRepository) {
   def validateOpiskeluoikeusChange(oldState: KoskeenTallennettavaOpiskeluoikeus, newState: KoskeenTallennettavaOpiskeluoikeus): HttpStatus = {
+    HttpStatus.fold(
+      validateOpiskeluoikeudenTyypinMuutos(oldState, newState),
+      validateLähdejärjestelmäIdnPoisto(oldState, newState),
+      validateOppilaitoksenMuutos(oldState, newState),
+      validateVanhanOpiskeluoikeudenTapaukset(oldState, newState, ePerusteet)
```

Comment:
Vähän huolettaa, tuleeko tästä mahdollisesti merkittäviä suorituskykyheikennyksiä, jos eperusteita aletaan nyt tutkia useammassa tiedonsiirrossa? Jos kerta aiemmin käytännössä validaatiota ei oltu vielä tehty.

Reply:
Tosin on ePerusteita näemmä käytetty aiemminkin muissa validoinneissa.

Reply:
Hmm, voisi ehkä lisätä jonkin ~5 minuutin cachen tuohon, ihan siltä varalta kun tulee näitä massakirjoituksia kuitenkin silloin tällöin.

Pitää myös varmaan vähän miettiä että mites jos eperusteet on alhaalla; onko se riittävä syy estää kokonaan opiskeluoikeuksien siirto.

Ehkä fiksuinta olisi kirjoittaa tämä muotoon että vasta jos on tehty mahdollisesti ongelmallisia muutoksia, niin sitten tarkistetaan että mitäs ne eperusteet sanoo.